src/api-binder.ts

```diff
@@ -549,18 +549,20 @@ export class APIBinder {
 				}
 
 				await this.report();
-				await Bluebird.delay(REPORT_SUCCESS_DELAY);
-				await this.reportCurrentState();
+				this.reportCurrentState();
```

Comment:
I cannot get what this recursive call does... Should not it be called in the error case only? It would cause the second attempt to report the state which most likely will not happen because of the zero diff. But what is the point? We call `report` on a state change anyway?

Reply:
I'm not sure actually, this is still left over from the typescript rewrite. I'm going to make some heavy architectural changes to this module soon so I'll look at it properly then.

Reply:
It's in case a new change has come in whilst we were checking the previous one, and if there's nothing to report then we just exit early. I expect there could be nicer/clearer ways to do it though